src/github.com/getlantern/flashlight/main/pac.go

```diff
@@ -20,22 +21,29 @@ import (
 )
 
 var (
-	isPacOn        = int32(0)
-	pacURL         string
-	muPACFile      sync.RWMutex
-	pacFile        []byte
-	directHosts    = make(map[string]bool)
-	shouldProxyAll = int32(0)
-	onepac         = &sync.Once{}
+	isPacOn     = int32(0)
+	pacURL      string
+	directHosts = make(map[string]bool)
+	cfgMutex    sync.RWMutex
 )
 
-func ServeProxyAllPacFile(b bool) {
-	if b {
-		atomic.StoreInt32(&shouldProxyAll, 1)
-	} else {
-		atomic.StoreInt32(&shouldProxyAll, 0)
+func ServePACFile() {
+	cfgMutex.Lock()
+	defer cfgMutex.Unlock()
+	if pacURL == "" {
+		pacURL = ui.Handle("/proxy_on.pac", http.HandlerFunc(sercePACFile))
+	}
+}
+
+func sercePACFile(resp http.ResponseWriter, req *http.Request) {
+	log.Trace("Serving PAC file")
+	resp.Header().Set("Content-Type", "application/x-ns-proxy-autoconfig")
+	resp.WriteHeader(http.StatusOK)
```

Comment:
For HTTP/1.1, Go automatically uses chunked transfer encoding if no content-length header is specified.  That begs the question if any user agents request the PAC file using HTTP/1.0.  Incidentally, the previous version of this code also did not include the content-length header for the PAC file.